.github/workflows/service_test_atomicdata.yml

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Service Test Atomicdata
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "core/src/**"
+      - "core/tests/**"
+      - "!core/src/docs/**"
+      - "!core/src/services/**"
+      - "core/src/services/atomicdata/**"
+      - ".github/workflows/service_test_atomicdata.yml"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+jobs:
+  atomicdata:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Test
+        shell: bash
+        working-directory: core
+        run: |
+          cargo test atomicdata --features services-atomicdata -j=1
+        env:
+          RUST_BACKTRACE: full
+          OPENDAL_ATOMICDATA_TEST: off
+          OPENDAL_ATOMICDATA_ROOT: /tmp/opendal
+          OPENDAL_ATOMICDATA_ENDPOINT: http://localhost:9883
+          OPENDAL_ATOMICDATA_PRIVATE_KEY: TODO - ${{ secrets.OPENDAL_ATOMICDATA_PRIVATE_KEY }}
```

Comment:
Looks like I don't have permissions to set up atomic-server and edit the key vault/secrets for github action's CI. So I  left these fields as TODO and test currently turned off. To test locally, one can simply:
- Install atomic-server locally following the steps at: [https://github.com/atomicdata-dev/atomic-server#4-install-using-cargo](https://github.com/atomicdata-dev/atomic-server#4-install-using-cargo)
- Start atomic-server by running `atomic-server` in terminal and configure the agent info
- Run `cargo test atomicdata --features services-atomicdata` in another tab under project root

Reply:
Hi, what's the `http://localhost:9883` point for? Seems there is not a server started in our workflow.

And if the atomic server is start locally, the public key and private key could be configured directly without need of github secrets.

Reply:
Sorry for the confusion. I left `OPENDAL_ATOMICDATA_ENDPOINT` in yml file as a configurable parameter for github actions.

To test locally I'm using `.env` file with the set of public/private keys specific to the atomic server on my machine. 

To be able to test it on github actions, it is probably needed to install atomic server on the testing server (not sure how it works), and then that instance of atomic-server will be able to generate its own specific set of public/private key for the agent.

Reply:
Looks good to me. The github action env is an ubuntu server, we can install atomic server on it.